__tests__/app.test.js

```diff
@@ -39,4 +39,59 @@ describe("GET /api", () => {
 				expect(endpoints).toEqual(endpointsData);
 			});
 	});
+	it("GET 200 - each endpoint has the correct key-value pairs", () => {
+		return request(app)
+			.get("/api")
+			.expect(200)
+			.then(({ body: { endpoints } }) => {
+				for (let endpoint in endpoints) {
+					if (endpoint !== "GET /api") {
+						expect(endpoints[endpoint].hasOwnProperty("description")).toBe(
+							true
+						);
+						expect(endpoints[endpoint].hasOwnProperty("queries")).toBe(true);
+						expect(endpoints[endpoint].hasOwnProperty("exampleResponse")).toBe(
+							true
+						);
+					}
+				}
+			});
+	});
+});
```

Comment:
This isn't really an accurate test. You can require the endpoints json file into the test file and test that the response you receive from the server looks exactly the same as the endpoints file you required in. This would be an accurate test

Reply:
Hi Rayhaan. I was wondering about this :/ as Niamh gave me the feedback before this test, based on my initial /api test below (I had required in the endpoints data etc and was comparing it as you've mentioned :D)

-----------------------
Perhaps an additional test - great code so far, really clean work :)

In [__tests__/app.test.js](https://github.com/Jared-Ha/be-nc-news/pull/2#discussion_r1677660995):

> @@ -29,3 +30,13 @@ describe("GET /api/topics", () => {
 			});
 	});
 });
+describe("GET /api", () => {
+	it("GET 200 - responds with a JSON object detailing all available endpoints", () => {
+		return request(app)
+			.get("/api")
+			.expect(200)
+			.then(({ body: { endpoints } }) => {
+				expect(endpoints).toEqual(endpointsData);
+			});
+	});
Lovely work here! I think it might be nice to add another test that checks they have the correct key-value pairs as well? :)
-------------------------